dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3191,7 +3191,7 @@ Sedp::Writer::write_participant_message(const ParticipantMessageData& pmd,
             (ser << pmd);
 
   if (ok) {
-      send_sample(payload, size, reader, sequence);
+    send_sample(payload, size + padding, reader, sequence, reader != GUID_UNKNOWN);
```

Comment:
If this uses `size+padding`, does line 3184 need to change as well?

Reply:
I think we're potentially banking on the fact that there's no padding in this message (is that true)? But yeah, either change it both places or don't change it in either place.

Reply:
Actually, there are 4 places like this, plus 3 places of new ACE_Message_Block(size). They were all changed to size + padding in my original draft PR#1952, together with DCPS::Message_Block_Ptr replacing DCPS::Message_Block and separate delete statements. Should I bring them in this PR?

Reply:
In write_stateless_message(), line 3221: new ACE_Message_Block(size + padding)); but line 3232: send_sample(payload, size, reader, sequence);